pkgs/development/libraries/gtk+/3.0-immodules.cache.patch

```diff
@@ -0,0 +1,24 @@
+--- a/gtk/deprecated/gtkrc.c	2016-04-02 18:43:08.401663112 +0900
++++ b/gtk/deprecated/gtkrc.c	2016-04-02 18:29:19.927608592 +0900
+@@ -771,5 +771,20 @@
+   const gchar *var = g_getenv ("GTK_IM_MODULE_FILE");
+   gchar *result = NULL;
+ 
++  // check NIX_PROFILES paths.
++  const gchar *nixProfilesEnv = g_getenv ("NIX_PROFILES");
++  gchar *cachePath;
++  guint i;
++
++  if(nixProfilesEnv){
++    gchar **paths = g_strsplit(nixProfilesEnv, " ", -1);
++    for (i = 0; paths[i] != NULL; i++){
++      cachePath = g_build_filename(paths[i], "etc", "gtk-3.0", "immodules.cache", NULL);
++      if( g_file_test( cachePath, G_FILE_TEST_EXISTS) ){
++        result = g_strdup(cachePath);
++      }
```

Comment:
> Thanks for the memory leak information, I added g_strfreev(paths); and g_free(cachePath);, is that enough to fix it?

Almost, there is still a leak when 1) there is more than one nix profile containing the immodule.cache file, and 2) if the environment variable GTK_IM_MODULE_FILE is set.

I think the easiest fix is to:

1. Move the NIX_PROFILES check below the original one and change the `if` to `if (nixProfilesEnv && !result)`.
2. Free the result variable before it is overwritten, i.e. `if (result) g_free(result)` right before the assignment.

That should fix all of the memory leaks.